qiime/plugin/testing.py

```diff
@@ -84,3 +89,34 @@ def assertSemanticTypeRegisteredToFormat(self, semantic_type, exp_format):
             obs_format, exp_format,
             "Expected semantic type %r to be registered to format %r, not %r."
             % (semantic_type, exp_format, obs_format))
+
+    def transform_format(self, source_format, target, filename=None,
+                         filenames=None):
+        # Guard any non-QIIME2 Format sources from being tested
+        self.assertTrue(issubclass(source_format, FormatBase))
```

Comment:
Sorry, missed this earlier: for the same reasons why we added the `ValueError` below, I think i makes sense to raise a `TypeError` here, instead of asserting the type. If a developer tries to use the wrong format with this helper, it shouldn't be a test failure, it should be a flat-out error. We could say something like  `source_format must be a subclass of FormatBase`, instead of listing out the three possible subclasses.